In [1]:
import pandas as pd

df = pd.read_csv("usarrests.csv")

print(df.shape)
display(df.head())

(50, 4)


,Unnamed: 0,Murder,Assault,UrbanPop
0,Alabama,13.2,236.0,58
1,Alaska,10.0,263.0,48
2,Arizona,8.1,294.0,80
3,Arkansas,8.8,190.0,50
4,California,9.0,276.0,91


In [2]:
df.describe()

,Murder,Assault,UrbanPop
count,50.00000,49.000000,50.00000
mean,7.78800,182.183673,74.20000
std,4.35551,130.877435,73.40828
min,0.80000,45.000000,6.00000
25%,4.07500,109.000000,53.25000
50%,7.25000,159.000000,66.00000
75%,11.25000,249.000000,77.75000
max,17.40000,879.000000,570.00000


In [3]:
df[df["UrbanPop"] > 100]

,Unnamed: 0,Murder,Assault,UrbanPop
14,Iowa,2.2,56.0,570


In [4]:
df.sort_values("Assault", ascending=False).head()

,Unnamed: 0,Murder,Assault,UrbanPop
39,South Carolina,14.4,879.0,48
32,North Carolina,13.0,337.0,45
8,Florida,15.4,335.0,80
19,Maryland,11.3,300.0,67
2,Arizona,8.1,294.0,80


In [5]:
df[df["Assault"].isna()]

,Unnamed: 0,Murder,Assault,UrbanPop
9,Georgia,17.4,NaN,60


Flagged Values

Iowa – UrbanPop = 570

This is impossible because UrbanPop is a percentage and it cannot be over 100%. I would check the original data due to the fact of it probably a mistake.

South Carolina – Assault = 879

This number is really high compared to the rest, but still possible. I would keep it for now and I would checkthe original source before changing anything.

Georgia – Assault is missing

What Georgia is missing is the assault value, which is why there are only 49 values instead of 50. I would check the original data before filling it in or removing the row.

GitHub Repository: 
https://github.com/leonhong2007/cs82a-portfolio/tree/main/module3